src/start-proxy-action.ts

```diff
@@ -198,7 +207,11 @@ function getCredentials(logger: Logger): Credential[] {
     "registries_credentials",
   );
   const registrySecrets = actionsUtil.getOptionalInput("registry_secrets");
-  const language = actionsUtil.getOptionalInput("language");
+  const languageString = actionsUtil.getOptionalInput("language");
+  const language = languageString ? parseLanguage(languageString) : undefined;
+  const registryTypeForLanguage = language
+    ? LANGUAGE_TO_REGISTRY_TYPE[language]
+    : undefined;
```

Comment:
nit: If you apply the suggestion above, then this becomes simpler:

```suggestion
  const registryTypeForLanguage = LANGUAGE_TO_REGISTRY_TYPE[language];
```

Reply:
I still need the check on `language` itself, though.